### ISSUE TITLE
From a form: Back to first page of list instead of previous page

### ISSUE THREAD

**Issue by youring** [odoo/odoo#7791]
We noticed this behaviour on all lists in Odoo 7/8/Master :
For example from partner list, go to second or third page of a list, then we click on an item to open the form.
Then we click on the breadcrumb to go back to partner list or switch to list view or click Back button of browser, then Odoo brings us to the first page of the list, which is very annoying.
Intended behavior is to bring back to the page we just came from.

Is this a bug or is it the "normal" behavior ?
Thanks in advance for your answers.

---copied from https://bugs.launchpad.net/openobject-addons/+bug/1280388
BTW. it seems fixed in an old version https://bugs.launchpad.net/openobject-client-web/+bug/524547

**Comment by hparfr** [comment]
+1

**Comment by intero-chz** [comment]
+1

There is one behaviour i don't like too while going back by browser or breadcrumb:
Grouped lists fully closed again...

**Comment by bealdav** [comment]
:+1:

**Comment by hparfr** [comment]
I asked Odoo to solve this bug under the terms of the Odoo Enterprise subscription.
Here is the response : 

>We do not consider this issue as a blocking bug. This is seen as a possible improvement, for future releases, but this won't be corrected in stable releases of Odoo.
>To know more about our stable policy, you can read the below documentation:
https://github.com/odoo/odoo/wiki/Contributing#what-does-stable-mean

This kind of small annoying UI bugs impacts the user experience of our clients.

> This is seen as a __possible__ improvement, for future releases

I hope this will be put on the roadmap. Already 2 years since this has bug been reported (on launchpad).

**Comment by dp-wabro** [comment]
+1

**Comment by codingforfun** [comment]
+1

**Comment by codingforfun** [comment]
This bug has been introduced with 19d9cc048140386c1133fd863d39a749f8c5ec57

**Comment by codingforfun** [comment]
I hacked something here: https://github.com/OCA/OCB/pull/475 
I feel its a bit ugly, also one problem remains, so please feel free to review and comment.

**Comment by Paasky** [comment]
+1 Similar bug when coming back to a list view using the breadcrumbs - ex: if you opened a partner from the 2nd page of a list, then click back using the breadcrumbs, you'll be thrown to the 1st page. Same behaviour in Odoo 9 Enterprise.

Our client is highly annoyed at this, any ideas when @codingforfun 's fix could be implemented?

**Comment by idncom** [comment]
+1 most commonly complained about issue from our clients.  Still exists in 9, and it's in both list and kanban views...doesn't seem like it would be that hard to fix, but codingforfun's solution is just for list.  Anyone know where a similar js solution for kanban could be applied?  There is not a js file for kanban in the same folder :(

**Comment by Yenthe666** [comment]
Hey all,

There have been quite a few of bug reports about this in the past.
This is actually a limitation of the framework and we don't consider this as a bug. This behaviour might be changed or improved in upcoming versions though.

Regards,
Yenthe

**Comment by codingforfun** [comment]
Hey. Is Odoo actually sometimes fixing some of the issues reported or is it the basic approach of the company to just close issues after 2 years of doing nothing? 
I mean this issue is something I would really consider a major issue. 
As I understand the whole purpose of Odoo it’s all about handling a massive amount of business data. Being not able to at least navigate a simple list of such massive data in a reasonable way is something really difficult to sell to a customer. The time back I wrote the linked PR I was a total JS noob and managed to find at least an acceptable improvement to this. So I wouldn’t consider this rocket science.  Telling us 

> This is actually a limitation of the framework and we don't consider this as a bug

is somewhat disappointing. Odoo is providing the framework, so this is nothing that just happened to you. You are in full control over this. 

Sorry. Don’t want to be rude, but after seeing issues one after another after a long time of ignoring contributions your solution & explanation is just plain disappointing.

**Comment by Yenthe666** [comment]
@codingforfun let me add some context here: I'm not working for Odoo, I do this on a voluntary basis in order to help and improve both Odoo and the issue tracker. If you look at issues from the last months you'll see the behaviour and response times have improved a lot.
I do not know the exact details of why it is (not yet) there but there was a limitation. Perhaps @mart-e or @ged-odoo can provide you with more details about the why, but I honestly cannot.

**Comment by ged-odoo** [comment]
@codingforfun We should have a special badge for @Yenthe666 , he is doing a lot of volunteer work, and takes the time to answer questions and triage issues. His help is really appreciated.

Now, back to your question, starting in saas-16, the list view now behaves like you want.  The change came with the new views. It definitely works in v11.  

For previous versions, the situation is not so clear.  While this is clearly annoying, this is not broken in the sense that the user data are destroyed, or some task is impossible to perform.  So, we try to avoid changing the behaviour of the user interface, especially if the fix is difficult and may cause issues.  The previous javascript code is very subtle/fragile, and a small change may cause other issues (a fix in a form view may break the navigation in the list view, of may not work in edit mode, or may cause a traceback when an onchange comes back with different values, ...). So I do not expect this behaviour to be changed in a stable version.

Also, I understand the frustration seeing issues/PR not even answered after 2 years.  But give me a chance to explain the situation.  Our JS team is about 4-8 persons, depending on the period.  We have >1000 issues opened on github, with 245 of them tagged framework. Most of these issues take more than 30 min of work, more frequently a few hours. We all have our internal work to do: work on new features, ...  There is no way we can keep up with that.  Not because we do not want to close all github issues, because we do, but because we need to do our work.

I think that we really improved our internal processes.  For example, the JS team has one day every week to work on various github issues.  Also, we focus much much much more on code quality, testing, documentation and bug fixes than before.  We even take some heat from the rest of Odoo because we do not implement new features as quickly as they would like, because we care. Also, we think a lot about various refactorings, to decrease our technical debt and make the code easier to maintain/test/extend.

In short, the message is: we really care, we invest all our available resources on code quality, but they are limited, and we go for the long term: we want the future to be bright, we want the v11, and v12, v13, ... to be bug free (as much as possible) and of high quality.

**Comment by codingforfun** [comment]
@Yenthe666 Sorry. Wasn’t clear to me you are not part of the Odoo team but doing voluntary work, because I thought this repo to be maintained by officials only. 

@ged-odoo Thanks for updating on this issue. Good to hear that it will be fixed in v11. Current version I work with is v10 and it was disappointing to see this existed over 4 major versions since v7. I know the JS code was very subtle/fragile and of course it’s not broken in the sense of loosing data, but the non-technical customer doesn’t care about the technical reasons if his subjective impression is to not being able to navigate and handle his data. In this subjective perspective this is nearly as bad as lost data. Because having data you can’t work with is basically useless.

**Comment by hparfr** [comment]
Thanks for these clarifications.

**Comment by Yenthe666** [comment]
Thanks a lot for the feedback from the technical side (and the positive words @ged-odoo). 😄 
@codingforfun no problem, you couldn't know. I'm quite an exception as I'm almost the only person in the world with these rights outside of Odoo. 😉 
Anyways, as @ged-odoo has answered it has been handled in 11 and up, so that is good news.
